MsixCore/msixmgr/msixmgr.cpp

```diff
@@ -481,44 +775,103 @@ int main(int argc, char * argv[])
                     std::wcout << std::endl;
 
                     OutputUnpackFailures(packageSourcePath, skippedFiles, failedPackages, failedPackagesErrors);
+
+                    // Telemetry : Workflow Log
+                    QueryPerformanceCounter(&msixMgrLoad_EndCounter);
+                    workflowElapsedTime = msixmgr::CalcWorkflowElapsedTime(msixMgrLoad_StartCounter, msixMgrLoad_EndCounter, msixMgrLoad_Frequency);
+                    msixmgr::TraceLogWorkflow(workflowId.c_str(), cli.GetOperationTypeAsString().c_str(), true, workflowElapsedTime, L"", L"");
                 }
             }
-            return hr;
+            return S_OK;
         }
         case OperationType::ApplyACLs:
         {
+            // Telemetry : ApplyACLs Workflow Log
+            msixmgr::TraceLogApplyACLsWorkflow(workflowId.c_str(), msixmgr::ExtractPackageNameFromFilePath(cli.GetPackageFilePathToInstall()).c_str());
+
+            HRESULT hrApplyACLs;
+
             std::vector<std::wstring> packageFolders;
             packageFolders.push_back(cli.GetPackageFilePathToInstall()); // we're not actually installing anything. The API just returns the file path name we need.
-            RETURN_IF_FAILED(MsixCoreLib::ApplyACLs(packageFolders));
+            hrApplyACLs = MsixCoreLib::ApplyACLs(packageFolders);
+
+            if (FAILED(hrApplyACLs))
+            {
+                errorCode = msixmgr::GetErrorCodeFromHRESULT(hrApplyACLs);
+                errorDesc = L"Failed ApplyACLs Operation. HRESULT " + errorCode + L".";
+
+                // Telemetry : Workflow Log
+                QueryPerformanceCounter(&msixMgrLoad_EndCounter);
+                workflowElapsedTime = msixmgr::CalcWorkflowElapsedTime(msixMgrLoad_StartCounter, msixMgrLoad_EndCounter, msixMgrLoad_Frequency);
+                msixmgr::TraceLogWorkflow(workflowId.c_str(), cli.GetOperationTypeAsString().c_str(), false, workflowElapsedTime, errorCode.c_str(), errorDesc.c_str());
+
+                RETURN_IF_FAILED(hrApplyACLs);
+            }
+
+            // Telemetry : Workflow Log
+            QueryPerformanceCounter(&msixMgrLoad_EndCounter);
+            workflowElapsedTime = msixmgr::CalcWorkflowElapsedTime(msixMgrLoad_StartCounter, msixMgrLoad_EndCounter, msixMgrLoad_Frequency);
+            msixmgr::TraceLogWorkflow(workflowId.c_str(), cli.GetOperationTypeAsString().c_str(), true, workflowElapsedTime, L"", L"");
+
             return S_OK;
         }
         case OperationType::MountImage:
         {
+            // Telemetry : Mount Workflow Log
+            msixmgr::TraceLogMountWorkflow(workflowId.c_str(), cli.GetFileTypeAsString().c_str());
+
             WVDFileType fileType = cli.GetFileType();
 
             if (cli.GetMountImagePath().empty())
             {
+                errorCode = msixmgr::GetErrorCodeFromHRESULT(E_INVALIDARG);
+                errorDesc = L"Mount Image Path Argument not given. HRESULT " + errorCode + L".";

```

Comment:
Same as above

Reply:
As discussed, all the Telemetry variables will have the same Error Message as the actual Console statements. The change for outputting the Error Message on Console from a variable will be done as part of Error Message Improvement Task that will be taken later.